src/psbt.c

```diff
@@ -2737,18 +2737,13 @@ int wally_finalize_psbt(struct wally_psbt *psbt)
             out_script_len = out.script_len;
         }
         if (input->witness_script) {
-            script = input->witness_script;
-            script_len = input->witness_script_len;
+            out_script = input->witness_script;
+            out_script_len = input->witness_script_len;
             witness = true;
         }
         if (input->redeem_script) {
```

Comment:
This if block can be swapped with the one above. Then the `if (type == WALLY_SCRIPT_TYPE_P2WSH...)` block can be dropped.